secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/sign/SapiJwsSigner.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.jws.sign;
+
+import java.util.Map;
+
+import org.forgerock.util.promise.Promise;
+
+/**
+ * Interface to implement custom signers
+ */
+public interface SapiJwsSigner {
+
+    /**
+     * Sign a Map<String, Object>, that represents a json structure, supporting critical header claims.<br/>
+     * @param payload              {@link Map}
+     * @param criticalHeaderClaims {@link Map}
```

Comment:
only critical header claims? Can we have a description please? How would you set kid for example?

Reply:
Just had to go looking in the code to find out that the 'DefaultSapiJwsSigner' adds the kid and alg by default. What if I wanted to add another default header parameter e.g. x5t or jwk? 

If that's not possible, that's OK, but the javadoc/api docs should state that so I don't spend ages looking at the code only to find it's impossible.

Reply:
We do not intend to implement a generic signer to cover all possible cases.
Maybe we need to rename this signer implementation, because has been implemented to cover the event signature case.
Comment: https://github.com/SecureApiGateway/secure-api-gateway-ob-uk/pull/429#discussion_r1348598610